blocks/header/header.js

```diff
@@ -308,31 +343,44 @@ export default async function decorate(block) {
     }
   });
 
+  const navWrapper = document.createElement('div');
+  navWrapper.className = 'nav-wrapper';
+  navWrapper.append(nav);
+  block.append(navWrapper);
+
+  navWrapper.addEventListener('mouseout', (e) => {
+    if (isDesktop.matches && !nav.contains(e.relatedTarget)) {
+      toggleAllNavSections(navSections);
+      overlay.classList.remove('show');
+    }
+  });
+
+  window.addEventListener('resize', () => {
+    navWrapper.classList.remove('active');
+    overlay.classList.remove('show');
+    toggleMenu(nav, navSections, false);
+  });
+
   // hamburger for mobile
   const hamburger = document.createElement('div');
   hamburger.classList.add('nav-hamburger');
   hamburger.innerHTML = `<button type="button" aria-controls="nav" aria-label="Open navigation">
       <span class="nav-hamburger-icon"></span>
     </button>`;
-  hamburger.addEventListener('click', () => toggleMenu(nav, navSections));
+  hamburger.addEventListener('click', () => {
+    navWrapper.classList.toggle('active');
+    overlay.classList.toggle('show');
+    toggleMenu(nav, navSections);
+  });
   nav.prepend(hamburger);
   nav.setAttribute('aria-expanded', 'false');
   // prevent mobile nav behavior on window resize
   toggleMenu(nav, navSections, isDesktop.matches);
   isDesktop.addEventListener('change', () => toggleMenu(nav, navSections, isDesktop.matches));
 
-  const navWrapper = document.createElement('div');
-  navWrapper.className = 'nav-wrapper';
-  navWrapper.append(nav);
-  block.append(navWrapper);
-
   renderAuthCombine(
     navSections,
     () => !isDesktop.matches && toggleMenu(nav, navSections, false),
   );
   renderAuthDropdown(navTools);
 }
-
-window.addEventListener('resize', () => {
-  window.location.reload();
-});
```

Comment:
@sirugh @loginesta Do you know why this was added?  just wanted to make sure once this is removed existing fix (if any) does not break.

Reply:
It was introduced in a previous (related) PR:
https://github.com/hlxsites/aem-boilerplate-commerce/pull/307/files

Should be safe to be removed!